OpenRA.Platforms.Default/Sdl2GraphicsContext.cs

```diff
@@ -0,0 +1,258 @@
+		{
+			switch (pt)
+			{
+				case PrimitiveType.PointList: return OpenGL.GL_POINTS;
+				case PrimitiveType.LineList: return OpenGL.GL_LINES;
+				case PrimitiveType.TriangleList: return OpenGL.GL_TRIANGLES;
+			}
+
+			throw new NotImplementedException();
+		}
+
+		public void DrawPrimitives(PrimitiveType pt, int firstVertex, int numVertices)
+		{
+			VerifyThreadAffinity();
+			OpenGL.glDrawArrays(ModeFromPrimitiveType(pt), firstVertex, numVertices);
+			OpenGL.CheckGLError();
+		}
+
+		public void Clear()
+		{
+			VerifyThreadAffinity();
+			OpenGL.glClearColor(0, 0, 0, 1);
+			OpenGL.CheckGLError();
+			OpenGL.glClear(OpenGL.GL_COLOR_BUFFER_BIT | OpenGL.GL_DEPTH_BUFFER_BIT);
+			OpenGL.CheckGLError();
+		}
+
+		public void EnableDepthBuffer()
+		{
+			VerifyThreadAffinity();
+			OpenGL.glClear(OpenGL.GL_DEPTH_BUFFER_BIT);
+			OpenGL.CheckGLError();
+			OpenGL.glEnable(OpenGL.GL_DEPTH_TEST);
+			OpenGL.CheckGLError();
+			OpenGL.glDepthFunc(OpenGL.GL_LEQUAL);
+			OpenGL.CheckGLError();
+		}
+
+		public void DisableDepthBuffer()
+		{
+			VerifyThreadAffinity();
+			OpenGL.glDisable(OpenGL.GL_DEPTH_TEST);
+			OpenGL.CheckGLError();
+		}
+
+		public void ClearDepthBuffer()
+		{
+			VerifyThreadAffinity();
+			OpenGL.glClear(OpenGL.GL_DEPTH_BUFFER_BIT);
+			OpenGL.CheckGLError();
+		}
+
+		public void SetBlendMode(BlendMode mode)
+		{
+			VerifyThreadAffinity();
+			OpenGL.glBlendEquation(OpenGL.GL_FUNC_ADD);
+			OpenGL.CheckGLError();
+
+			switch (mode)
+			{
+				case BlendMode.None:
+					OpenGL.glDisable(OpenGL.GL_BLEND);
+					break;
+				case BlendMode.Alpha:
+					OpenGL.glEnable(OpenGL.GL_BLEND);
+					OpenGL.CheckGLError();
+					OpenGL.glBlendFunc(OpenGL.GL_ONE, OpenGL.GL_ONE_MINUS_SRC_ALPHA);
+					break;
+				case BlendMode.Additive:
+				case BlendMode.Subtractive:
+					OpenGL.glEnable(OpenGL.GL_BLEND);
+					OpenGL.CheckGLError();
+					OpenGL.glBlendFunc(OpenGL.GL_ONE, OpenGL.GL_ONE);
+					if (mode == BlendMode.Subtractive)
+					{
+						OpenGL.CheckGLError();
+						OpenGL.glBlendEquation(OpenGL.GL_FUNC_REVERSE_SUBTRACT);
+					}
+
+					break;
+				case BlendMode.Multiply:
+					OpenGL.glEnable(OpenGL.GL_BLEND);
+					OpenGL.CheckGLError();
+					OpenGL.glBlendFunc(OpenGL.GL_DST_COLOR, OpenGL.GL_ONE_MINUS_SRC_ALPHA);
+					OpenGL.CheckGLError();
+					break;
+				case BlendMode.Multiplicative:
+					OpenGL.glEnable(OpenGL.GL_BLEND);
+					OpenGL.CheckGLError();
+					OpenGL.glBlendFunc(OpenGL.GL_ZERO, OpenGL.GL_SRC_COLOR);
+					break;
+				case BlendMode.DoubleMultiplicative:
+					OpenGL.glEnable(OpenGL.GL_BLEND);
+					OpenGL.CheckGLError();
+					OpenGL.glBlendFunc(OpenGL.GL_DST_COLOR, OpenGL.GL_SRC_COLOR);
+					break;
+			}
+
+			OpenGL.CheckGLError();
+		}
+
+		public void Dispose()
+		{
+			if (disposed)
+				return;
+
+			disposed = true;
+			if (context != IntPtr.Zero)
+			{
+				SDL.SDL_GL_DeleteContext(context);
+				context = IntPtr.Zero;
+			}
+
+			SDL.SDL_Quit();
```

Comment:
https://github.com/OpenRA/OpenRA/issues/15188 suggests `SDL.SDL_Quit` needs to live in the window layer, not GL.

Reply:
Good catch. I forgot to delete this when copying and editing the dispose method - it was actually defined in both places.